sdp/numpy_fft_sdp.py

```diff
@@ -14,7 +14,8 @@ def sliding_dot_product(Q, T, order="F"):
     tmp = np.empty((2, shape), order=order)
     tmp[0, :m] = Q[::-1]
     tmp[0, m:] = 0.0
-    tmp[1, :] = T
+    tmp[1, :n] = T
+    tmp[1, n:] = 0.0
     fft_2d = np.fft.rfft(tmp, axis=-1)
 
-    return np.fft.irfft(np.multiply(fft_2d[0], fft_2d[1]))[m - 1 : n]
+    return np.fft.irfft(np.multiply(fft_2d[0], fft_2d[1]), n=shape)[m - 1 : n]
```

Comment:
Just to recap, here we've made two changes in this file:

(1) fixing `tmp[1, :]=T`
(2) passing `n=shape` when calling irfft. 

With new tests, I can see that:
* without (1) & (2): `test_sdp_case2`, `test_sdp_case3`, and `test_sdp_case4` all fail.
* with only (1): `test_sdp_case2` fails
* with only (2): `test_sdp_case3` and `test_sdp_case4` fail.